cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1701,15 +1701,33 @@ timestamp LocalVersionedEngine::latest_timestamp(const std::string& symbol) {
 }
 
 struct AtomicKeySizesInfo {
-    size_t count;
+    size_t count{0};
     std::atomic<size_t> compressed_size;
     std::atomic<size_t> uncompressed_size;
 };
 
+using KeySizeCalculators = std::vector<std::pair<VariantKey, stream::StreamSource::ReadContinuation>>;
+
+static void read_ignoring_key_not_found(Store* store, KeySizeCalculators&& calculators) {
+    if (calculators.empty()) {
+        return;
+    }
+
+    std::vector<folly::Future<folly::Unit>> res;
```

Comment:
I think it'll be slightly more efficient to pull out `store->batch_read_compressed(std::move(calculators), BatchReadArgs{})` in a variable and then pre-allocate res using reserve.